### PR TITLE
Replace in-place links with buttons

### LINKED ISSUE #1990: Multiple components use "in place links" instead of buttons
**Describe the bug**
First noticed this on Items with many Bitstreams: the pagination buttons of the simple file section trigger an unnecessary page reload when clicked.
* This is due to the fact those buttons are `<a [routerLink]="[]">`
* If we use `<button>` instead, this won't happen (and a11y would be improved)
* There's at least three similar cases (Community list, metadata representation, related Items)

**To Reproduce**
1. Find or create an Item with more than 5 Bitstreams in the ORIGINAL Bundle 
   (e.g.: https://demo7.dspace.org/handle/10673/244311)
2. Go to the simple Item page
3. In the file section, click on "Show more" / "Collapse"
   * Each click will cause the whole page to reload
   * Pagination still works though

**Expected behavior**
Buttons that don't do any routing should not have router links.

**Related work**
https://github.com/DSpace/dspace-angular/pull/1567: One of the automatic upgrades changed these from `<a [routerLink]="">` to `<a [routerLink]="[]">` but at the time I didn't notice the bug

### PR DESCRIPTION
**Pull request by LotteHofstede**
## References
* Fixes #1990 

## Description
This PR replaces instances of empty/in-place links with buttons

## Instructions for Reviewers
Compare the following links/buttons before and after this PR:
- The "View more" button on the /community-list page
- The "View more" and "Collapse" links on the item page for the download list
- The "View more" and "View less" links on the publication page for the author (metadata-representation) list
- The "View more" and "View less" links on the item page for the related items list
- The search bar magnifying glass icon in the header

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by tdonohue — CHANGES_REQUESTED** [review]
@LotteHofstede : Overall the changes look good.  But, I'm noticing that the searchbox in the header is now misaligned...and the search icon in the header seems to have moved further away from the globe (just more whitespace between them).

search-nav-alignment

I'm testing in production mode (`yarn build:prod; yarn serve:ssr`).

I thought it might be a cached stylesheet at first, but tried in several browsers (Chrome and Firefox) and see the same in both.  I also did a complete rebuild (including a `yarn clean`) and still see the same behavior.

**Comment by LotteHofstede** [comment]
@tdonohue Thanks for the review!
I was able to reproduce the issue, so it definitely wasn't your cache!

The issue was caused by the fact that I renamed a custom CSS class but forgot to change it in the HTML for that particular instance. It should be fixed now!

image

File: src/app/search-navbar/search-navbar.component.html [hunk 1]
```diff
@@ -4,9 +4,9 @@
       <input #searchInput [@toggleAnimation]="isExpanded" [attr.aria-label]="('nav.search' | translate)" name="query"
              formControlName="query" type="text" placeholder="{{searchExpanded ? ('nav.search' | translate) : ''}}"
              class="d-inline-block bg-transparent position-absolute form-control dropdown-menu-right p-1" [attr.data-test]="'header-search-box' | dsBrowserOnly">
-      <a class="submit-icon" [routerLink]="" (click)="searchExpanded ? onSubmit(searchForm.value) : expand()" [attr.data-test]="'header-search-icon' | dsBrowserOnly">
+      <button class="submit-icon btn btn-link btn-link-inline" type="button" (click)="searchExpanded ? onSubmit(searchForm.value) : expand()" [attr.data-test]="'header-search-icon' | dsBrowserOnly">
```
Comment (tdonohue) [inline src/app/search-navbar/search-navbar.component.html:7]: I'm getting an accessibility error here now (from the axe DevTools scanner plugin in Chrome) which says "Buttons must have discernible text".  The issue appears to be that this has no text within it, or `aria-label` or `title`.  

Could we add a `title` or `aria-label` attribute here?  That should fix the accessibility issue.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@LotteHofstede : Looks good now overall!  And the CSS issues are fixed on my end.

However, I found one last (very minor) accessibility issue.  We have a new `<button>` that has no text to describe it.  It should be quick to fix (see comment below).

File: src/app/search-navbar/search-navbar.component.html (hunk 1 above)
Comment (LotteHofstede) [reply src/app/search-navbar/search-navbar.component.html]: Added aria-label

**Review by tdonohue — APPROVED** [review]
👍 Thanks @LotteHofstede !  This looks good to me now & verified it works and passes accessibility checks.

**Pull request merged**